### DISCUSSION TITLE
Location of Pixel Data Tag in DICOM File

### DISCUSSION THREAD

**Discussion by chrisbll971** [Enet4/dicom-rs#540] [Q&A]
Hello,

I was wondering if there is a way to find the exact index position of the pixel data within a DICOM file? For example, if the DICOM file is read into a Vec<u8>, the pixel data might start at the index 934 in the vector. Or alternatively, is there is a way to only read each frame into memory 1-by-1 vs. reading the entire pixel data into memory? The use case is we have some very large uncompressed files that we are trying to read and process (up to 80 GB+).

Thank you

**Comment by Enet4** [comment]
Thank you for bringing this up. Requiring objects to reside completely in memory is a known issue, but there are plans for an API which would enable reading bigger data sets in workable chunks.

Until then, the alternative available is to use the mid-level [data set reader API](https://docs.rs/dicom-parser/latest/dicom_parser/dataset/read/index.html) in the `dicom_parser` crate. This API fetches DICOM data in tokens, and I believe would also let you peek the byte offset of the current position in the DICOM file. Working with this API will naturally be a bit trickier than the object API, but hopefully the next major versions will incorporate the aforementioned lazy/partial reading API.
